### PR TITLE
Add model providers and Supported functions

### PR DESCRIPTION
**Pull request by adarshpalaskar1**
This commit addresses issue #121, adding a new doc page with a table for model providers vs supported functions.

### CODE REVIEW

**Comment by adarshpalaskar1** [comment]
Sorry for the delayed PR. Please let me know if this is the table format you are looking for.

* I'm unsure if the catch-all implementation is correct and if LocalServerOpenAISchema should be included.
* Haven't included the quick explanation of the functions yet, how should they differ from the overview mentioned here(https://siml.earth/PromptingTools.jl/dev/how_it_works#ai*-Functions-Overview)? I think an example would be helpful.

Please let me know about the required changes.

**Comment by svilupp** [comment]
This is great! Thank you for pooling this together.

On the function descriptions, you're right. It would be duplicative, but it would also keep the relevant information together for reference. I don't feel strongly either, so pick whichever you think is better for a new user.

> I'm unsure if the catch-all implementation is correct and if LocalServerOpenAISchema should be included.
I like how you added the footnotes. I'd probably mention that LocalServerOpenAISchema is specifically for seamless integration for Llama.jl. In general, it's different only by using the ENV variable for URL (making the integration easier in some workflows, eg, when you have nested calls and passing api_kwargs is harder).

File: docs/src/model_providers_and_supported_functions.md
Comment (svilupp) [inline docs/src/model_providers_and_supported_functions.md:11]: I'm not sure how to add it, but for the aiscan/aiimage functions, I think it would be good to mention that they rely on the specific endpoints being implemented by the provider.

Comment (svilupp) [inline docs/src/model_providers_and_supported_functions.md:11]: Why does aiimage have only one tick vs aiscan has many? Aren't they the same logic / dispatch?

Comment (svilupp) [inline docs/src/model_providers_and_supported_functions.md:11]: If it's not too much effort, I think it would look visually much better to have aiextract be the 3rd column... Because it's much more ticks / more supported.

Unless it's way too much manual effort

File: docs/src/model_providers_and_supported_functions.md
Comment (svilupp) [inline docs/src/model_providers_and_supported_functions.md:20]: See the discussion about difference in Ollama schemas explained here: https://github.com/svilupp/PromptingTools.jl/issues/68

I think it would be great to add which endpoint they hit, eg, `Ollama (endpoint `api/chat`)` or something like that. 

What do you think?

File: docs/src/model_providers_and_supported_functions.md
Comment (svilupp) [inline docs/src/model_providers_and_supported_functions.md:28]: Why is the P capitalized? Is it because of Preferences.jl or a typo?

File: docs/make.jl
Comment (svilupp) [inline docs/make.jl:50]: Could we please make it shorter? It will stretch the sidebar width.

Maybe "Coverage of Model Providers"? Any other ideas?

**Review by svilupp — CHANGES_REQUESTED** [review]

File: docs/src/model_providers_and_supported_functions.md
Comment (adarshpalaskar1) [reply docs/src/model_providers_and_supported_functions.md:11]: Would a footnote below the table work? Or do you think there will be too many of them eventually?

File: docs/src/model_providers_and_supported_functions.md
Comment (adarshpalaskar1) [reply docs/src/model_providers_and_supported_functions.md:28]: A typo😅. Should I change it to lowercase?

File: docs/src/model_providers_and_supported_functions.md
Comment (svilupp) [reply docs/src/model_providers_and_supported_functions.md]: I don't feel strongly. 

I think it would work well for now - we don't have that many footnotes yet and we can always change it later!

Comment (svilupp) [reply docs/src/model_providers_and_supported_functions.md]: Yes, I believe that's correct (it's fuzzy because of the model vs provider vs schema difference, but it's a pretty good summary).

Comment (svilupp) [reply docs/src/model_providers_and_supported_functions.md]: Just mark it as resolved when you're done.

File: docs/make.jl
Comment (svilupp) [reply docs/make.jl]: I like both! Sizewise it's not that different, so pick your favourite.

Or it could be "Provide Coverage by Function" (meant as ai* functions), but I'll leave that to you.

Or "Model Provide Coverage" (to avoid the "of")

**Comment by svilupp** [comment]
@adarshpalaskar1 Would you mind please:
- adding the GroqOpenAI schema? It supports only aigenerate, aiextract
- marking the resolved items as resolved? I don't want to untick them until I know you're done, so it's easier if you could mark it.

**Comment by adarshpalaskar1** [comment]
> * adding the GroqOpenAISchema? It supports only aigenerate, aiextract
Yes, will do that.

> * marking the resolved items as resolved? I don't want to untick them until I know you're done, so it's easier if you could mark it.
Sure, I was just waiting for your confirmation on the changes.

File: docs/make.jl
Comment (adarshpalaskar1) [reply docs/make.jl]: I think your initial suggestion, "Coverage of Model Providers", would be the easiest to understand. So maybe we should go with that.

**Review by svilupp — APPROVED** [review]

**Comment by svilupp** [comment]
Let me know when it’s ready. It looks good to me!

**Comment by adarshpalaskar1** [comment]
> Let me know when it’s ready. It looks good to me!

This should be ready to be merged😀

**Comment by svilupp** [comment]
Awesome stuff! Thank you for contributing, @adarshpalaskar1 !

**Pull request merged**
